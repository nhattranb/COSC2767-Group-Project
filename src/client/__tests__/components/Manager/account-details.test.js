/*
 *   RMIT University Vietnam
 *   Course: COSC2767 Systems Deployment and Operations
 *   Semester: 2024C
 *   Assessment: Assignment 2
 *   Author: Le Dinh Ngoc Quynh
 *   ID: s3791159
 *   Created  date: 21 Dec 2024
 *   Last modified: 21 Dec 2024
 */

import React from 'react';
import { render } from '@testing-library/react';
import AccountDetails from "../../../app/components/Manager/AccountDetails";
import {expect, jest, test} from "@jest/globals";

test('AccountDetails component snapshot', () => {
    const user = {
        provider: 'Email',
        email: 'test@example.com',
        firstName: 'John',
        lastName: 'Doe',
        phoneNumber: '1234567890'
    };

    const accountChange = jest.fn();
    const updateProfile = jest.fn();

    const { asFragment } = render(
        <AccountDetails user={user} accountChange={accountChange} updateProfile={updateProfile} />
    );

    expect(asFragment()).toMatchSnapshot();
});
