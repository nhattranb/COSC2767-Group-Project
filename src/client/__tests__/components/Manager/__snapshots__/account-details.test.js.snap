// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetails component snapshot 1`] = `
<DocumentFragment>
  <div
    class="account-details"
  >
    <div
      class="info"
    >
      <div
        class="desc"
      >
        <p
          class="one-line-ellipsis mr-3"
        >
          test@example.com
        </p>
        <span
          class="custom-badge custom-badge-secondary"
          style="border-radius: 3px;"
        >
          Member
        </span>
      </div>
    </div>
    <form>
      <div
        class="row"
      >
        <div
          class="col-12 col-md-6"
        >
          <div
            class="input-box "
          >
            <label>
              First Name
            </label>
            <div
              class="input-text-block"
            >
              <input
                autocomplete="on"
                class="input-text"
                name="firstName"
                placeholder="Please Enter Your First Name"
                type="text"
                value="John"
              />
            </div>
            <span
              class="invalid-message"
            />
          </div>
        </div>
        <div
          class="col-12 col-md-6"
        >
          <div
            class="input-box "
          >
            <label>
              Last Name
            </label>
            <div
              class="input-text-block"
            >
              <input
                autocomplete="on"
                class="input-text"
                name="lastName"
                placeholder="Please Enter Your Last Name"
                type="text"
                value="Doe"
              />
            </div>
            <span
              class="invalid-message"
            />
          </div>
        </div>
        <div
          class="col-12 col-md-12"
        >
          <div
            class="input-box "
          >
            <label>
              Phone Number
            </label>
            <div
              class="input-text-block"
            >
              <input
                autocomplete="on"
                class="input-text"
                name="phoneNumber"
                placeholder="Please Enter Your Phone Number"
                type="text"
                value="1234567890"
              />
            </div>
            <span
              class="invalid-message"
            />
          </div>
        </div>
      </div>
      <hr />
      <div
        class="profile-actions"
      >
        <button
          class="input-btn custom-btn-secondary md text-only icon-left "
          style="border-radius: 3px;"
          type="submit"
        >
          <span
            class="btn-text"
          >
            Save changes
          </span>
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
